captum/attr/_core/deep_lift.py

```diff
@@ -669,6 +715,11 @@ def nonlinear(module, delta_in, delta_out, grad_input, grad_output, eps=1e-10):
         grad_input[0],
         grad_output[0] * delta_out[0] / delta_in[0],
     )
+
+    # If the module is invalid, save the newly computed gradients
+    # The original_grad_input will be overridden later in the Tensor hook
+    if module.is_invalid:
+        module.saved_grad = grad_input[0]
```

Comment:
I think it might be better to not have this override here and instead update _check_valid_module to use the grad_fn of inputs at the prehook. Currently, this change would stop catching new issues when _check_valid_module is not true (mismatch of inputs and input gradients), and this may occur as we add more modules which use this method?

I think this can be done by storing the grad_fn attributes from the inputs in prehook and using that in _check_valid_module as opposed to the inputs provided in the hook itself. The advantage of this approach is that it would do the appropriate checks for any inplace module, not necessarilly only those with this nonlinear backward override method.